### DISCUSSION TITLE
Welcome to Openwrt-R2S Discussions!

### DISCUSSION THREAD

**Discussion by quintus-lab** [quintus-lab/OpenWRT-Rockchip#7] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Mikemansour** [comment]
Hello, I am a noobie  and I really want to know how to install this OpenWRT, 
I already have the r4s nanopi and I could install friendlywrt on the SD card but it doesn't support many features 
I want to know how to compile the full image of openwrt together and create a bootable image on the TF card, Is there any guide that you guys can recommend to me? 
Thanks in advance

**Reply by quintus-lab** [reply]
https://openwrt.org/docs/guide-user/additional-software/beginners-build-guide

**Comment by Prog-5** [comment]
Hi, I have got recently a nanopi r4s.
I installed the R4S-QC-2021-03-26-sfs release and everything is fine, but when I want to install nfs-kernel-server or ffmpeg, I get "kernel version incompatible" error: "The installed version of package kernel is not compatible, require 5.4.105-1-c2d5c7a8… while 5.4.102-1-39c4bbe0… is installed."
I tried many versions of other days' releases (K510s, too) and I get alike kernel version incompatible errors.

Could you please help me how to install the nfs and ffmpeg package, or modify the daily release to have the same kernel version required.

Thanks in advance

**Reply by quintus-lab** [reply]
Could you list all the packages which you want to install?
Iet me  check which kmod dependencies need to be added.

BTW, there is highly recommended you use 21.02 with kernel 5.4[2102](https://github.com/quintus-lab/OpenWRT-Rockchip/releases/tag/2102)

**Reply by quintus-lab** [reply]
for nfs-kernel-server
CONFIG_PACKAGE_kmod-crypto-cts=y
CONFIG_PACKAGE_kmod-fs-nfs-common=y
CONFIG_PACKAGE_kmod-fs-exportfs=y
CONFIG_PACKAGE_kmod-fs-nfs-common-rpcsec=y
CONFIG_PACKAGE_kmod-fs-nfsd=y
CONFIG_PACKAGE_kmod-dnsresolver=y
CONFIG_PACKAGE_kmod-fs-nfs=y
CONFIG_PACKAGE_kmod-fs-nfs-v4=y

for ffmpeg:
CONFIG_PACKAGE_kmod-sound-core=y
CONFIG_PACKAGE_kmod-input-core=y

you cant add that into .config then compile your owner OpenWRT image.

**Reply by Prog-5** [reply]
It seems that only next to "kernel" is the incompatible message, the other dependencies meet the requirements to install:
 
Details for package nfs-kernel-server
Version: 2.5.2-1
Size: ~171.9 KB installed
Dependencies:
libwrap (12.7 KB) NOT INSTALLED
libblkid1 INSTALLED
libuuid1 INSTALLED
librt INSTALLED
libpthread INSTALLED
libgcc1 INSTALLED
libtirpc INSTALLED
kmod-fs-nfsd (152.1 KB) NOT INSTALLED
kernel VERSION INCOMPATIBLE
kmod-fs-nfs-common (159.5 KB) NOT INSTALLED
kmod-fs-exportfs (233 B) NOT INSTALLED
kmod-fs-nfs-common-rpcsec (48.8 KB) NOT INSTALLED
kmod-crypto-des INSTALLED
kmod-crypto-cbc (2.2 KB) NOT INSTALLED
kmod-crypto-manager INSTALLED
kmod-crypto-aead INSTALLED
kmod-crypto-null INSTALLED
kmod-crypto-hash INSTALLED
kmod-crypto-pcompress INSTALLED
kmod-crypto-cts (3.6 KB) NOT INSTALLED
kmod-crypto-md5 INSTALLED
kmod-crypto-sha1 (2.4 KB) NOT INSTALLED
kmod-crypto-hmac INSTALLED
kmod-crypto-ecb INSTALLED
kmod-crypto-arc4 INSTALLED
kmod-fs-nfs (90.6 KB) NOT INSTALLED
kmod-dnsresolver (5.4 KB) NOT INSTALLED
kmod-fs-nfs-v4 (94.7 KB) NOT INSTALLED
rpcbind (29.5 KB) NOT INSTALLED
nfs-utils-libs (17.7 KB) NOT INSTALLED
libkeyutils1 (5.8 KB) NOT INSTALLED
libdevmapper-normal (148.0 KB) NOT INSTALLED | libdevmapper-selinux (149.1 KB) NOT INSTALLED
kmod-dm INSTALLED
kmod-dax INSTALLED
libselinux (67.4 KB) NOT INSTALLED
libsepol (270.3 KB) NOT INSTALLED
libpcre INSTALLED
musl-fts INSTALLED



Details for package ffmpeg
Version: 4.3.1-1
Size: ~95.9 KB installed
Dependencies:
libpthread INSTALLED
libgcc1 INSTALLED
libffmpeg-full (6.1 MB) NOT INSTALLED
zlib INSTALLED
libbz21.0 INSTALLED
alsa-lib (402.8 KB) NOT INSTALLED
kmod-sound-core (182.4 KB) NOT INSTALLED
kernel VERSION INCOMPATIBLE
kmod-input-core (104 B) NOT INSTALLED
librt INSTALLED
libgnutls INSTALLED
libnettle8 INSTALLED
libgmp10 INSTALLED
libatomic1 INSTALLED
libopus (183.0 KB) NOT INSTALLED
lame-lib (128.7 KB) NOT INSTALLED
fdk-aac (311.7 KB) NOT INSTALLED

I'm new to it, could you please, describe how can I add that into .config and then compile my own OpenWRT image?

Thanks, I will use 21.02 with kernel 5.4.

**Reply by quintus-lab** [reply]
fork the project, 
switch the 21.02 branche to default branche.
 
then edit the seed file: [rockchip.seed](https://github.com/quintus-lab/OpenWRT-Rockchip/blob/master/seed/rockchip.seed),
put the following line into rockchip.seed
#for nfs-kernel-server
CONFIG_PACKAGE_nfs-kernel-server=y
CONFIG_PACKAGE_kmod-crypto-cts=y
CONFIG_PACKAGE_kmod-fs-nfs-common=y
CONFIG_PACKAGE_kmod-fs-exportfs=y
CONFIG_PACKAGE_kmod-fs-nfs-common-rpcsec=y
CONFIG_PACKAGE_kmod-fs-nfsd=y
CONFIG_PACKAGE_kmod-dnsresolver=y
CONFIG_PACKAGE_kmod-fs-nfs=y
CONFIG_PACKAGE_kmod-fs-nfs-v4=y
#for ffmpeg:
CONFIG_PACKAGE_ffmpeg=y
CONFIG_PACKAGE_kmod-sound-core=y
CONFIG_PACKAGE_kmod-input-core=y

modify the github actions yml file [openwrt-rockchip-2102.yml](https://github.com/quintus-lab/OpenWRT-Rockchip/blob/21.02/.github/workflows/openwrt-rockchip-2102.yml) to suit your needs
